dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/guestbook/GuestbookPage.java

```diff
@@ -238,54 +240,50 @@ public String save() {
             }
         }
 
-        Command<Dataverse> cmd;
-        try {
-            if (editMode == EditMode.CREATE || editMode == EditMode.CLONE) {
-                guestbook.setCreateTime(new Timestamp(new Date().getTime()));
-                guestbook.setUsageCount(new Long(0));
-                guestbook.setEnabled(true);
-                dataverse.getGuestbooks().add(guestbook);
-                cmd = new UpdateDataverseCommand(dataverse, null, null, dvRequestService.getDataverseRequest(), null);
-                commandEngine.submit(cmd);
-                create = true;
-            } else {
-                cmd = new UpdateDataverseGuestbookCommand(dataverse, guestbook, dvRequestService.getDataverseRequest());
-                commandEngine.submit(cmd);
-            }
+        if (editMode == EditMode.CREATE || editMode == EditMode.CLONE) {
+            Try.of(() -> guestbookService.saveGuestbook(guestbook))
+                    .onSuccess(dv -> JsfHelper.addFlashMessage(BundleUtil.getStringFromBundle("guestbook.create")))
+                    .onFailure(this::handleErrorMessages);
+        } else {
+            Try.of(() -> guestbookService.editGuestbook(guestbook))
+                    .onSuccess(dv -> JsfHelper.addFlashMessage(BundleUtil.getStringFromBundle("guestbook.save")))
+                    .onFailure(this::handleErrorMessages);
+        }
+
```

Comment:
This won't work the same as before, in catch they were returning null which for redirecting is the same as empty string. You have to use `if(tryVariable.isFailure()) return ""`.